lib/internal/Magento/Framework/Module/Dir.php

```diff
@@ -45,6 +45,10 @@ public function getDir($moduleName, $type = '')
     {
         $path = $this->componentRegistrar->getPath(ComponentRegistrar::MODULE, $moduleName);
 
+        if (! isset($path)) {
+            throw new \LogicException("Component '$moduleName' of type '$type' is not correctly registered.");
```

Comment:
This change is not backward compatible, you're adding new exception type that could be thrown. I think it would be better to use ```\InvalidArgumentException``` in order to keep backward compatibility there.